lib/model/CTokenListCategory.cc

```diff
@@ -380,8 +396,19 @@ CTokenListCategory::TSizeSizePr CTokenListCategory::orderedCommonTokenBounds() c
 }
 
 std::size_t CTokenListCategory::maxMatchingStringLen() const {
-    // Add a 10% margin of error
-    return (m_MaxStringLen * 11) / 10;
+    // Add a 10% margin of error if this wouldn't result in a length much
+    // longer than the base string.  The broader the category (i.e. fewer
+    // tokens that must match) the lower the tolerance of length increases.
+    // The risk is that we end up with a category with just one or two tokens
+    // that matches every message.  The max matching length is designed to
+    // prevent categories that just require a few tokens to match from
```

Comment:
Wouldn't this be desirable over time? Its effectively category collapsing/generalization which is desired.

Reply:
Up to a point it's desirable, but not when you have one category that's defined by just one or two tokens that ends up matching _everything_.

The situation I observed (in a VMware ESXi log) is that every message contains the hostname, so that gives one token that is present in every single message.  Then "Hostd" is also present in a lot of messages, some of which have lots of other tokens after that and others of which have just a single brace or number.  These messages that have the hostname and "Hostd" as their only tokens end up in a category together.  Then longer, more informative messages end up in different categories.  But over time, messages are seen that fit within the 10% tolerance of the hostname and "Hostd" category.  Gradually this increases its max matching string length until it dominates all the longer categories.  You end up with a category with just two tokens that has hundreds of other categories in its `preferred_to_categories`.

I will send you the file that can reproduce this problem before the change this PR makes.

Reply:
Gotcha. 

So, we are restricting the growth, but not preventing it altogether. Makes sense, especially for these potentially all encompassing categories.